src/lib/operators/join_sort_merge.cpp

```diff
@@ -638,6 +645,53 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractJoinOperatorImpl {
     return new_pos_list;
   }
 
+  void _apply_additional_predicates(PosList& left, PosList& right) {
+    DebugAssert(left.size() == right.size(), "Left and right table should have the same size.");
+
+    if (!_additional_join_predicates.has_value()) {
+      return;
+    }
+
+    if (_additional_join_predicates.value().empty()) {
+      return;
+    }
+
+    PosList filtered_left;
+    filtered_left.resize(left.size());
+    PosList filtered_right;
+    filtered_right.resize(right.size());
```

Comment:
Have you thought about using `reserve()` instead of `resize()`? `resize()` will create default-inserted elements and not only reserve more memory. After the loop, when using `reserve()`, you can then shrink the reserved memory to fit only actually inserted elements by calling `shrink_to_fit()`.

Reply:
This code will be removed anyway.
The reason we used resize is because we override every value in the following loop. We can now use the [] operator to overwrite the values instead of push_back. Matkus said this is faster.

Reply:
A quick benchmark seems to indicate the opposite, but maybe the benchmark I used is a bit too simple: http://quick-bench.com/n9PZycApfAqy6F-wkPhF0t7TyZQ
![n9pzycapfaqy6f-wkphf0t7tyzq](https://user-images.githubusercontent.com/2145092/52785034-4803b000-3057-11e9-8b70-ffdefd2a9194.png)

Reply:
Try again with integers. vec.emplace_back("foo"); might bis faster because we save a copy operation?

Reply:
I tried it with `RowID`s now and resizing really is much faster. Good to know! 
![bkmjfahgwgn-vavdnbra8fdfmie](https://user-images.githubusercontent.com/2145092/52786265-bd24b480-305a-11e9-84d8-f2e1a9fc3399.png)
http://quick-bench.com/bkmJfAhGWGN-vAvdNBRa8FDFmIE